### PR TITLE
cmake: Export CVC4 library interface.

### PR DESCRIPTION
**Pull request by mpreiner**
The `find_package` configuration information is now generated by CVC4's build system.

This allows cmake projects that use CVC4 to search for CVC4 libraries via  `find_package(CVC4)` and link CVC4 and all its dependencies via `target_link_libraries(<target> CVC4::cvc4)` instead of defining a custom finder module.

### CODE REVIEW

**Comment by 4tXJ7f** [comment]
Hmm, from a cursory look at Travis, it looks like making `target_include_directories` `PRIVATE` does not play well with the parser because the parser is not part of the CVC4 library but needs access to the definition of floating-point numbers. Any idea of how we could fix this? I suppose that long-term, we don't want src/util/floatingpoint.h to be part of the public API anymore, which would fix the problem.

**Comment by mpreiner** [comment]
Well for now we have to mark all includes that are exported to the public API as `PUBLIC`, all others can be `PRIVATE`. As soon as the new API is default we can make all others also `PRIVATE`.

**Comment by mpreiner** [comment]
Seems like the only dependencies that get exported to the public API are GMP, CLN and SymFPU.

File: CMakeLists.txt [hunk 1]
```diff
@@ -481,6 +481,26 @@ if(BUILD_BINDINGS_JAVA OR BUILD_BINDINGS_PYTHON)
...
+#-----------------------------------------------------------------------------#
+# Package configuration
+#
+# Install config and configversion for CVC4 for find_package(CVC4).
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/CVC4ConfigVersion.cmake
+  VERSION ${CVC4_RELEASE_STRING}
+  COMPATIBILITY SameMajorVersion
```
Comment (4tXJ7f) [inline CMakeLists.txt]: Hmm, isn't it more like `SameMinorVersion` for us? E.g. for 1.7 we [changed the API a bit](https://github.com/CVC4/CVC4/blob/eba03c5af0112aea04d83977333ae37e8a13137d/NEWS#L39-L40).

File: CMakeLists.txt [hunk 2]
```diff
@@ -481,6 +481,26 @@ if(BUILD_BINDINGS_JAVA OR BUILD_BINDINGS_PYTHON)
...
+)
+
+# Install the config, configversion and custom find modules
+install(FILES
+  ${CMAKE_CURRENT_LIST_DIR}/cmake/CVC4Config.cmake
```
Comment (4tXJ7f) [inline CMakeLists.txt]: Why aren't we using `configure_package_config_file()` to generate the real `CVC4Config.cmake` file from a template as [recommended](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#command:configure_package_config_file)?

File: CMakeLists.txt (hunk 1 above)
Comment (mpreiner) [reply CMakeLists.txt]: Yep, changed.

File: CMakeLists.txt (hunk 2 above)
Comment (mpreiner) [reply CMakeLists.txt]: Changed it to use `configure_package_config_file()`. Tested it with another cmake project and `find_package(CVC4)` works as expected.

**Comment by mpreiner** [comment]
Yeah, that's a good idea!

**Pull request merged**